### PR TITLE
chore: update recipe for 5.0.2

### PR DESCRIPTION
**Pull request by agoose77**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.
* [ ] Restore test without xrootd

### CODE REVIEW

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@chrisburr I took the liberty of doing a naive upgrade to uproot5. I noticed that we are currently using an `uproot-base` output to split `uproot` into a metapackage. I'm not a Conda expert, so bear with me, but is my understanding correct that we do this to make the main uproot package noarch? 

Could you help me to understand the benefit of this split (I'm sure there is one, but I can't immediately recognise it)? It would seem to me that the naive single-noarch package should work.

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@henryiii could you cast your eye over this if you get a minute?

**Comment by agoose77** [comment]
@conda-forge-admin, please rerende

**Comment by agoose77** [comment]
conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,17 +25,14 @@ requirements:
     - zstandard
     - python-xxhash
     # Other common requirements
-    - xrootd  # [not win]
```
Comment (henryiii) [inline recipe/meta.yaml:26]: I thought the idea around removing this was so it could be noarch?

Comment (henryiii) [reply recipe/meta.yaml:26]: noarch will cause it to only test on linux. To the best of my knowledge, there's still no workaround for adding other platforms for testing.

**Comment by henryiii** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
@conda-forge-admin, please rerender

**Comment by agoose77** [comment]
OK! I learned some things here!

It looks like the `$PYTHON` variable isn't expanded when using `outputs`. I assume this is related to the [note](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section) in `conda-build`. Other packages seem to have seen this too: [conda-forge/staged-recipes@`a7c7559` (#13756)](https://github.com/conda-forge/staged-recipes/pull/13756/commits/a7c75594b4c5469da755cd69752e23feca684597)
I've opted to just use `python` and let `$PATH` resolve it. If that's a bad approach, then apparently the solution is to expand `PYTHON` in a binary e.g. create a `build.sh`.

**Comment by agoose77** [comment]
Merging this so that at least we have something on conda-forge.

**Pull request merged**